model/src/main/kotlin/com/android/gradle/replicator/model/internal/DefaultAndroidResourcesInfo.kt

```diff
@@ -73,15 +78,58 @@ class AndroidResourcesAdapter: TypeAdapter<AndroidResourcesInfo>() {
     override fun write(output: JsonWriter, value: AndroidResourcesInfo) {
         output.beginObject()
         for (resourceType in value.resourceMap) {
-            val folderObject = output.name(resourceType.key).beginArray()
+            output.name(resourceType.key).beginArray()
             for (resourceProperties in resourceType.value) {
-                val resourceObject = folderObject.beginObject()
-                resourceObject.name("qualifiers").value(resourceProperties.qualifiers)
-                resourceObject.name("extension").value(resourceProperties.extension)
-                resourceObject.name("quantity").value(resourceProperties.quantity)
-                resourceObject.endObject()
+                output.beginObject()
+                output.name("qualifiers").value(resourceProperties.qualifiers)
+                output.name("extension").value(resourceProperties.extension)
+                output.name("quantity").value(resourceProperties.quantity)
+                when (resourceProperties.propertyType) {
+                    ResourcePropertyType.VALUES -> {
+                        output.name("valuesFileList").beginArray()
+                        (resourceProperties as AndroidValuesResourceProperties).valuesMapPerFile.forEach { resourceFile ->
```

Comment:
shouldn't this serialization of the sub type be delegated to another class ?

Reply:
It's hard to separate them since the classes are tightly wound and share a bunch of code. I've moved the code that serializes and unserializes individual resource categories into a separate adapter in ResourceData, and separated the specialized parts into different methods. That should make this code more modular.